packages/studio-ui/src/hooks/useInjectUserStyles.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect } from "react";
+import useStudioStore from "../store/useStudioStore";
+
+export const UserCustomStyletagAttribute = "studio-user-custom-style";
+
+export default function useInjectUserStyles(
+  iframeDocument: Document | undefined
+) {
+  const [componentTree, getComponentMetadata, UUIDToFileMetadata, pageCss] =
+    useStudioStore((store) => [
+      store.actions.getComponentTree(),
```

Comment:
Chatted offline, we computed the list of CSS files from the component tree and only subscribe the `useEffect` to the CSS files instead of the component tree.